File: frontend/Concord/src/api/notificationsService.js
import { request } from './httpClient';
import { buildPagingQuery } from './pagination';

export async function getNotifications({ page, pageSize } = {}) {
  return request('Notifications', { query: buildPagingQuery(page, pageSize) });
}

export async function markNotificationRead(notificationId) {
  await request(`Notifications/${notificationId}:Read`, { method: 'POST' });
}

export async function markAllNotificationsRead() {
  await request('Notifications/Read', { method: 'POST' });
}

File: mobile/concord/lib/api/notifications_service.dart
import 'api_client.dart';
import 'models/notification_response.dart';
import 'models/paged_result.dart';

class NotificationsService {
  NotificationsService(this._client);

  final ApiClient _client;

  Future<PagedResult<NotificationResponse>> list({int page = 1, int pageSize = 30}) async {
    final data = await _client.get('/Notifications', query: {'page': page, 'pageSize': pageSize});
    return PagedResult.fromJson(data as Map<String, dynamic>, NotificationResponse.fromJson);
  }

  Future<void> markRead(String notificationId) => _client.post('/Notifications/$notificationId:Read');

  Future<void> markAllRead() => _client.post('/Notifications/Read');
}
